src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -103,18 +104,18 @@ private void AppendValue(ViewBufferValue value)
         private ViewBufferPage GetCurrentPage()
         {
             ViewBufferPage page;
-            if (Pages.Count == 0)
+            if (_pages.Count == 0)
             {
                 page = new ViewBufferPage(_bufferScope.GetPage(_pageSize));
-                Pages.Add(page);
+                _pages = _pages.Add(page);
```

Comment:
We shouldn't need the assignment here or any of these places since you're just mutating the struct in place